astroquery/eso/core.py

```diff
@@ -661,6 +661,12 @@ def retrieve_data(self, datasets, cache=True, continuation=False):
                 filename = self._request("GET", fileLink, save=True,
                                          continuation=True)
                 files.append(system_tools.gunzip(filename))
+        # Empty the redirect cache of this request session
+        # Only available and needed for requests versions < 2.17
+        try:
+            self._session.redirect_cache.clear()
+        except:
```

Comment:
can we catch a specific exception type here?

Reply:
@keflavich Done in the following commit.